### DISCUSSION TITLE
Make my QR codes smaller / Case Insensitive

### DISCUSSION THREAD

**Discussion by WPS12TA** [YOURLS/YOURLS#3688] [Q&A] [answered]
I am running YOURLS on my APACHE server. I installed it in sub directory /u/ 
I have used the case insensitive plugin to make the short codes insensitive.
I have installed a symlink between /U/ and /u/ that works.
I want to try and make an **option** of the WHOLE url being case insensitive **for a QR code**. So that 
MYSERVER.COM/U/ABC in a QR code
will go to ...
https://myserver.com/u/abc 
But all lower case can be written by default too.
Everything works as desired except the conversion of the /U/ to /u/
Can anyone help me please?
The reason for this is that making the url upper case makes for effective shortening in a QR code as upper case is encoded more efficiently than lower 5.5 : 8 I think.

**Comment by ozh** [comment]
I don't think changing the case will make a QR code smaller. Anyway, this should be done at the server level, with a .htaccess rule. See https://www.google.com/search?client=firefox-b-d&q=htaccess+redirect+to+lower+case for starters

**Comment by WPS12TA** [comment]
It does shorten a QR code (read the question). I'm hoping for a working
.htaccess modification or server modification to give upper and lower case
working alternatives rather than a promotion to try Google (I have failed
to progress through that route).

**Reply by dgw** [reply]
> It does shorten a QR code (read the question).

You also said "I think" as part of that assertion.

> rather than a promotion to try Google (I have failed to progress through that route).

With [this ServerFault Q&A](https://serverfault.com/a/1043764/13440) as the first result for @ozh's suggested query… you didn't make any progress by searching the web? At any rate, this is more of an Apache question than a YOURLS question. We aren't Apache experts here.

Since it sounds like the problematic part is `/u`, I suggest the alternative solution of moving YOURLS to a subdomain, say `u.myserver.com` instead of `myserver.com/u`. Same character count, but eliminates any need to mess with the request URI casing at all.

**Comment by WPS12TA** [comment]
That's a great idea. Thanks for the lateral thinking. I'll post again if I
can get that working. Configuring a url to be case insensitive sounds so
easy but it has been defeating my efforts thus far.

**Comment by WPS12TA** [comment, marked as answer]
My problem is now solved. (I could not configure the sub-domain method
correctly.)

To try and solve the problem first, I had created a symbolic link from the
U directory to u as a way of trying to get case insensitivity. The link
itself seemed to work in the file manager.

However, this seemed to STOP the following command working in the .htaccess
file in the root directory!

RewriteRule   ^U/(.*)   /u/$1   [R=301]

It took ages for me to establish this was issue! Since I could map from /X/
and /x/ to /u/  so I unlinked the symbolic link from U to u and Voila!

Now I can use both the following equivalently

https://example.com/u/yourls

And

EXAMPLE.COM/U/YOURLS

with the latter being much better for a smaller QR code and now being
rewritten to the one above.

Thanks to all who tried to help.

**Reply by WPS12TA** [reply]
Upper case Alpha characters do store better in QR codes than lower.
